.github/workflows/check-extra-os.yaml

```diff
@@ -55,20 +55,12 @@ jobs:
           make
           sudo make install
 
-      - uses: r-lib/actions/setup-r-dependencies@v1
+      - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-           extra-packages: rcmdcheck
+           extra-packages: any::rcmdcheck
 
-      - uses: r-lib/actions/check-r-package@v1
+      - run: rappdirs::user_cache_dir()
+        shell: Rscript {0}
```

Comment:
This still doesn't work - maybe try replacing this with:
```
run: Sys.setenv("R_USER_CACHE_DIR" = rappdirs::user_cache_dir())
```
:question: I think that `Rscript{0}` commands are now run in the same shell process (that is, without implicit `$SHELL`) , so that envvars set that way are now carried over to subsequent steps:grey_question:

Reply:
This was just for debugging to confirm the issue with the wrong path separator also happens on the github runner and checking the log it does! So this can also be removed if you want.

Reply:
Great, thanks, but can you maybe just try setting the envvar to see whether that works?

Reply:
Sure we can try but I think we won't get around using `fs` to fix it :D

Reply:
thanks!